test/y2partitioner/widgets/menus/add_test.rb

```diff
@@ -0,0 +1,267 @@
+      expect(subject.items).to include(item_with_id(:menu_add_btrfs))
+    end
+
+    it "includes 'Bcache'" do
+      expect(subject.items).to include(item_with_id(:menu_add_bcache))
+    end
+
+    it "includes 'Partition'" do
+      expect(subject.items).to include(item_with_id(:menu_add_partition))
+    end
+
+    it "includes 'Logical Volume'" do
+      expect(subject.items).to include(item_with_id(:menu_add_lv))
+    end
+  end
+
+  describe "#disabled_items" do
+    context "when the device is a disk device (Hard Disk, BIOS RAID, Multipath, DASD)" do
+      let(:scenario) { "one-empty-disk.yml" }
+
+      let(:device_name) { "/dev/sda" }
+
+      it "contains 'Logical Volume'" do
+        items = [:menu_add_lv]
+        expect(subject.disabled_items).to contain_exactly(*items)
+      end
+    end
+
+    context "when the device is a partition" do
+      let(:scenario) { "mixed_disks.yml" }
+
+      let(:device_name) { "/dev/sda1" }
+
+      it "contains 'Logical Volume'" do
+        items = [:menu_add_lv]
+        expect(subject.disabled_items).to contain_exactly(*items)
+      end
+    end
+
+    context "when the device is a MD RAID" do
+      let(:scenario) { "md_raid.yml" }
+
+      let(:device_name) { "/dev/md/md0" }
+
+      it "contains 'Logical Volume'" do
+        items = [:menu_add_lv]
+        expect(subject.disabled_items).to contain_exactly(*items)
+      end
+    end
+
+    context "when the device is a LVM Volume Group" do
+      let(:scenario) { "trivial_lvm.yml" }
+
+      let(:device_name) { "/dev/vg0" }
+
+      it "contains 'Partition'" do
+        items = [:menu_add_partition]
+        expect(subject.disabled_items).to contain_exactly(*items)
+      end
+    end
+
+    context "when the device is a LVM Logical Volume" do
+      let(:scenario) { "trivial_lvm.yml" }
+
+      let(:device_name) { "/dev/vg0/lv1" }
+
+      it "contains 'Partition'" do
+        items = [:menu_add_partition]
+        expect(subject.disabled_items).to contain_exactly(*items)
+      end
+    end
+
+    context "when the device is a Bcache" do
+      let(:scenario) { "bcache1.xml" }
+
+      let(:device_name) { "/dev/bcache0" }
+
+      it "contains 'Logical Volume'" do
+        items = [:menu_add_lv]
+        expect(subject.disabled_items).to contain_exactly(*items)
+      end
+    end
+
+    context "when the device is a Btrfs" do
+      let(:scenario) { "trivial_btrfs.yml" }
+
+      let(:device_name) { "/dev/sda1" }
+
+      subject { described_class.new(device.blk_filesystem) }
+
+      it "contains 'Partition' and 'Logical Volume'" do
+        items = [:menu_add_partition, :menu_add_lv]
+        expect(subject.disabled_items).to contain_exactly(*items)
+      end
+    end
+  end
+
+  describe "#handle" do
+    shared_examples "no action" do |scenario, device_name|
+      context "and another type of device is selected" do
+        let(:scenario) { scenario }
+
+        let(:device_name) { device_name }
+
+        it "calls none action" do
```

Comment:
NP: "none" looks weird here. I would simply say "calls no action". But it's not a big deal.